db/pgstorage/pgstorage.go

```diff
@@ -308,9 +313,12 @@ func (p *PostgresStorage) GetLatestL1SyncedExitRoot(ctx context.Context, dbTx pg
 		exitRootNum int64
 		exitRoots   [][]byte
 	)
-	const getLatestL1SyncedExitRootSQL = "SELECT block_id, global_exit_root_num, global_exit_root, exit_roots FROM syncv2.exit_root ORDER BY global_exit_root_num DESC LIMIT 1"
+	const getLatestL1SyncedExitRootSQL = "SELECT block_id, global_exit_root_num, global_exit_root, exit_roots FROM syncv2.exit_root WHERE block_id IS NOT NULL ORDER BY global_exit_root_num DESC LIMIT 1"
 	err := p.getExecQuerier(dbTx).QueryRow(ctx, getLatestL1SyncedExitRootSQL).Scan(&ger.BlockID, &exitRootNum, &ger.GlobalExitRoot, pq.Array(&exitRoots))
 	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return nil, gerror.ErrStorageNotFound
+		}
 		return nil, err
 	}
```

Comment:
The changed code works, but as a suggestion for sure we can improve the indentation level with the suggested code.

But the `suggested code` has a `bug` and it will return `nil, err` even if `err` is `nil`, to fix that we should add the `err != nil` clause to the `else` path, like this:

```
	if errors.Is(err, pgx.ErrNoRows) {
		return nil, gerror.ErrStorageNotFound
	} else if err != nil {
		return nil, err
	}
```

Improvements:
- Indentation level reduced by 1, better code readability
- Checks performed are reduced by 1 when an error happens

Reply:
you are totally right. Thank you